src/containers/User/Login.tsx

```diff
@@ -65,82 +62,122 @@ const Login: React.FC = () => {
   }
 
   useEffect(() => {
-    if (loginPayload?.orgList?.length === 0) {
+    if (loginPayload?.orgList?.length === LOGIN_AS_NO_ORG) {
       // No orgs, so skip org login
       finishLogin(loginPayload)
       return
     }
-    if (loginPayload?.orgList?.length === 1) {
-      // Auto-login with only one organisation
-      handleSubmit()
-    }
   }, [loginPayload])
 
-  const handleSelection = (e: any, data: any) => {
-    const { value } = data
-    setSelectedOrgIndex(value)
-  }
+  useEffect(() => {
+    // Organisation login
+    if (!loginPayload) return
+    const orgId = selectedOrgId
+    if (orgId === 0) {
+      // Log in without organisation
+      finishLogin(loginPayload)
+      return
+    }
+    attemptLoginOrg({
+      orgId,
+      JWT: loginPayload.JWT,
+      onLoginOrgSuccess: (loginOrgResult: LoginPayload) => {
+        finishLogin(loginOrgResult)
+      },
+    }).catch((error) => {
+      setNetworkError(error.message)
+    })
+  }, [selectedOrgId])
```

Comment:
Can't be done included in the previous `useEffect`, remember we were trying to avoiding having more than one of these per module

Reply:
You mean with both `loginPayload` and `selectedOrgId` in the same dependency array?

How would I then check for which one changed?

Reply:
I think it is anything that isn't dealt as a loginPayload change? I mean, you add the code for `selectedOrgId` after the code done for `loginPayload`. Just so it's concise in the same `useEffect`, like the portal to any changes on the current state of the login.